src/com/jwetherell/algorithms/mathematics/Division.java

```diff
@@ -59,6 +76,10 @@ public static final long divisionUsingShift(int a, int b) {
         int absB = Math.abs(b);
         int tempA, tempB, counter;
 
+        if (b == 0) {
```

Comment:
The method will throw an exception by ensuring the value of b before the result "long" is calculated avoiding unexpected results.